src/main/java/com/conveyal/taui/controllers/WorkerController.java

```diff
@@ -146,32 +150,33 @@ private Object singlePoint(Request request, Response response) {
         LOG.info("Re-issuing HTTP request from UI to worker at {}", workerUrl);
         HttpPost httpPost = new HttpPost(workerUrl);
         // httpPost.setHeader("Accept", "application/x-analysis-time-grid");
-        httpPost.setHeader("Accept-Encoding", "gzip"); // TODO copy all headers from request? Is this unzipping and re-zipping the result?
+        // TODO Explore: is this unzipping and re-zipping the result from the worker?
+        httpPost.setHeader("Accept-Encoding", "gzip");
         HttpEntity entity = null;
         try {
-            // Serialize and send the R5-specific task (not the one the UI sends to the broker)
+            // Serialize and send the R5-specific task (not the original one the broker received from the UI)
             httpPost.setEntity(new ByteArrayEntity(JsonUtil.objectMapper.writeValueAsBytes(task)));
             HttpResponse workerResponse = httpClient.execute(httpPost);
+            // Mimic the status code sent by the worker.
             response.status(workerResponse.getStatusLine().getStatusCode());
-            // TODO Should we exactly mimic these headers coming back from the worker?
-            response.header("Content-Type", "application/octet-stream");
-            response.header("Content-Encoding", "gzip");
```

Comment:
This header is necessary for the browser to automatically unzip.

Reply:
Removal of this line must have been a mistake. Nonetheless the situation here is a little more complicated than just adding a header - this actually causes the backend to unzip then rezip the body. Until I can find a workaround for it I guess it will have to work that way, but unzipping, buffering, and re-zipping in the backend is probably adding significant lag to these round trips.

Reply:
Added headers back in in e91e1a1